chapters/06-smart-contracts-in-ralph.md

```diff
@@ -1570,37 +1572,39 @@ async function test() {
   }
 
   await asset.transact.burn({
-     signer,
-     tokens: [{ id: asset.contractId, amount: 1n }]
+    signer,
+    tokens: [{ id: asset.contractId, amount: 1n }]
   })
-  const balance = await getBalance(signer.address)
-  console.assert(balance.tokenBalances?.[0].amount === "99")
+  const signerBalance = await getBalance(signer.address)
+  console.assert(signerBalance.tokenBalances?.[0].amount === '99')
 
   await asset.transact.lock({
-     signer,
-     args: { amount: 1n },
-     attoAlphAmount: DUST_AMOUNT,
-     tokens: [{ id: asset.contractId, amount: 1n }]
+    signer,
+    args: { amount: 1n },
+    attoAlphAmount: DUST_AMOUNT,
+    tokens: [{ id: asset.contractId, amount: 1n }]
   })
-  const balance2 = await getBalance(signer.address)
-  console.assert(balance2.tokenBalances?.[0].amount === "99")
-  console.assert(balance2.lockedTokenBalances?.[0].amount === "1")
+  const signerBalance2 = await getBalance(signer.address)
+  console.assert(signerBalance2.tokenBalances?.[0].amount === '99')
+  console.assert(signerBalance2.lockedTokenBalances?.[0].amount === '1')
 
   await asset.transact.deposit({
     signer,
     args: { amount: 1n },
     tokens: [{ id: asset.contractId, amount: 1n }]
   })
-  const balance3 = await getBalance(signer.address)
-  console.assert(balance3.tokenBalances?.[0].amount === "98")
+  const signerBalance3 = await getBalance(signer.address)
+  console.assert(signerBalance3.tokenBalances?.[0].amount === '98')
+  const contractBalance = (await asset.fetchState()).asset.tokens?.[0]?.amount
+  console.assert(contractBalance === 1n)
```

Comment:
I added an assertion for the contract state to demonstrate that the `deposit` call took 1 token from the signer and moved it to the contract state. Let me know if you find this useful!